File: system-test/sec-tests/typescript/sec_28_oauth.ts
// SEC-28: OAuth Security Defects
// Test: TC-CR-034
function buildOAuthUrl(clientId: string, redirectUri: string): string {
  return `https://auth.example.com/oauth2/authorize?response_type=token&client_id=${clientId}&redirect_uri=${redirectUri}`;
  // Missing: state parameter, PKCE, using implicit grant
}

File: system-test/sec-tests/java/Sec28OAuth.java
// SEC-28: OAuth Security Defects
// Test: TC-CR-034
package com.example;

public class OAuthService {

    public String buildAuthorizationUrl(String clientId, String redirectUri) {
        // Missing state parameter, implicit grant (response_type=token)
        return "https://auth.example.com/oauth2/authorize"
            + "?response_type=token"
            + "&client_id=" + clientId
            + "&redirect_uri=" + redirectUri
            + "&scope=openid";
    }
}
